wazuh/wazuh-dashboard-plugins
# Sync initial filters with app state

### Description
Hi team, 
this pull request aims to sync the initial state of the filters with the app URL state so that when the user reloads the page the filters persist.
 
### Issues Resolved
- #6918 

### Evidence
![Peek 2024-08-20 16-57](https://github.com/user-attachments/assets/d642e568-fdc3-4127-8f41-ffac581868f1)


### Test

<details>
<summary> Reload page in Threat hunting > Events </summary>

- Change the default date range in the search bar
- Add a normal filter and a pinned filter
- Reload the page (ctrl + r; f5)
- Check the filters persist

</details>

<details>
<summary> Reload page in Threat hunting > Dashboard </summary>

- Change the default date range in the search bar
- Add a normal filter and a pinned filter
- Reload the page (ctrl + r; f5)
- Check the filters persist

</details>

<details>
<summary> Navigate between apps </summary>

- Go to Threat hunting (for example)
- Change the default date range in the search bar
- Add a normal filter and a pinned filter
- In the main navigation menu select another app with dashboards
- Check the time filter persist
- Check the fields filters are gone

</details>

### Check List
